documentation/src/main/asciidoc/performance_guide/performance_guide.asciidoc

```diff
@@ -0,0 +1,232 @@
+= Infinispan {infinispanversion} Performance Guide
+The Infinispan community
+:toc2:
+:icons: font
+:toclevels: 3
+:numbered:
+:stem:
+
+This guide will give you information and tweaks about tuning Infinispan performance (both server and library mode).
+
+== Capacity planning
+
+Infinispan stores data in memory. Depending on operating mode data is stored in serialized form (server mode) or directly
+directly in memory (library mode). Data size can be estimated using sophisticated tools like  http://openjdk.java.net/projects/code-tools/jol/[Java Object Layout]
+and total amount of required memory can be calculated using the following formulas:
+
+[stem]
+.Total Data Set in library mode
+++++
+Total Data Set = Number Of Entries * (Key Size + Value Size + 200 b (Me tadata))
+++++
+
+[stem]
+.Total Data Set in server mode
+++++
+Total Data Set = Number Of Entries * (Serialized Key Size + Serialized Value Size + 200 b (Me tadata))
+++++
+
+
+In case of Local or Replicated mode, all data needs to fit in memory, so calculating the amount of required memory is trivial.
+
+Calculating memory requirements for Distributed mode is slightly more complicated and requires using the following:
+
+[stem]
+.Required memory in Distributed mode
+++++
+Required Memo ry = (Total Data Set)/(Nodes)*(1+(Node Failures)/(Nodes - Node Failures))
+++++
+
+Where:
+
+* `Total Data Set` - Estimated size of all data
+* `Nodes` - The number of nodes in the cluster
+* `Node Failures` - Number of possible failures (also `number of owners - 1`)
+
+Calculated amount of memory should be used for setting `Xmx` and `Xms` parameters.
+
+JVM as well as Infinispan require additional memory for other tasks like searches, allocating network buffers etc.
+It is advised to allocate no more than 50% of memory with living data.
+
+When considering large heaps, make sure there's enough CPU to perform garbage collection efficiently.
```

Comment:
I disagree with you here... I would like to avoid people creating VMs (e.g. using OpenStack) with 32GB or RAM and 1 vCPU for example. In that case GC would be terribly slow. 

Also, you often can not change memory bandwidth... But on the other hand you can [measure it](https://specs.openstack.org/openstack/nova-specs/specs/liberty/approved/mem-bw.html).